src/averageAges.js

```diff
@@ -37,7 +45,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(woman => woman.sex
+      === 'f' && people.some(person => person.mother === woman.name))
+    : people.filter(woman => woman.sex === 'f');
+  const sumOfAges
+  = women.reduce((sum, person) => sum + (person.died - person.born), 0);
```

Comment:
Check the code style
```suggestion
  const sumOfAges = women.reduce(
    (sum, person) => sum + (person.died - person.born), 
    0
  );
```